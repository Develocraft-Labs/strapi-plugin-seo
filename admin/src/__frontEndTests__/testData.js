import trads from "../translations";

export const id = 1,
  firstname = "John",
  lastname = "Doe",
  email = "John.Doe@companyA.com",
  title = "title",
  created_by = {
    id,
    email,
    firstname,
    lastname,
  },
  created_at = "2021-10-12T11:49:01.042Z",
  updated_at = "2021-10-15T09:43:36.344Z",
  updated_by = { email, firstname, id, isActive: true, lastname },
  seo = {
    canonical: "https://www.companyA.com/articles",
    created_at: "2021-10-12T11:55:00.595Z",
    created_by: 1,
    id: 1,
    jsonLd: {},
    metaDescription: "Artists Desc",
    metaTitle: "Artists Title",
    ogImage: "Artist og image",
    ogTitle: "Artists Og Title",
    ogType: "Artists Og Type",
    ogUrl: "Artists og url",
    title: "Artists Seo",
    updated_at,
    updated_by: 1,
  },
  resourceData = {
    title,
    id,
    created_by,
  },
  collectionName = "Artists",
  pagination = { page: 1, total: 13, pageSize: 5, pageCount: 3 },
  uid = "application::artists.artists",
  fullResults = [
    { created_at, created_by, id, seo, title, updated_at, updated_by },
    { created_at, created_by, id: 2, seo, title, updated_at, updated_by },
    { created_at, created_by, id: 3, seo, title, updated_at, updated_by },
    { created_at, created_by, id: 4, seo, title, updated_at, updated_by },
    { created_at, created_by, id: 5, seo, title, updated_at, updated_by },
    { created_at, created_by, id: 6, seo, title, updated_at, updated_by },
    { created_at, created_by, id: 7, seo, title, updated_at, updated_by },
    { created_at, created_by, id: 8, seo, title, updated_at, updated_by },
    { created_at, created_by, id: 9, seo, title, updated_at, updated_by },
    { created_at, created_by, id: 10, seo, title, updated_at, updated_by },
    { created_at, created_by, id: 11, seo, title, updated_at, updated_by },
    { created_at, created_by, id: 12, seo, title, updated_at, updated_by },
    { created_at, created_by, id: 13, seo, title, updated_at, updated_by },
  ],
  seoId = 1,
  apiID = "artists",
  attributes = {
    seo: {
      dominant: false,
      nature: "oneWay",
      plugin: "seo",
      private: false,
      target: "plugins::seo.seo",
      unique: false,
    },
    title,
  },
  schema = {
    attributes,
    collectionName,
    connection: "default",
    description: "",
    draftAndPublish: true,
    kind: "collectionType",
    name: "Artists",
    pluginOptions: {},
    restrictRelationsTo: null,
    visible: true,
  },
  resourceSchema = { collectionName, schema },
  pluginUID = "plugins::seo.seo",
  projectCollectionTypes = [{ apiID, schema, uid }],
  selectedSeo = {
    canonical: "https://www.companyA.com/articles",
    created_at: "2021-10-12T11:55:00.595Z",
    created_by: created_by,
    id: 1,
    jsonLd: {},
    metaDescription: "Artists Desc",
    metaTitle: "Artists Title",
    ogImage: "Artist og image",
    ogTitle: "Artists Og Title",
    ogType: "Artists Og Type",
    ogUrl: "Artists og url",
    title: "Artists Seo",
    updated_at,
    updated_by: updated_by,
  },
  refreshOptions = () => {},
  metaTitle = "meta title",
  metaDescription = "meta description",
  canonical = "canonical",
  handleDeleteSeo = () => {},
  handleEditSeo = () => {},
  defaultLocale = "en",
  userEnabledLocales = [
    {
      code: "en",
      created_at: "2021-12-20T09:03:53.724Z",
      id: 1,
      isDefault: true,
      name: "English (en)",
      updated_at: "2021-12-20T09:03:53.724Z",
    },
  ],
  seos = [seo, fullResults];

export const resourceComponentTestData = {
  resourceData,
  resourceSchema,
};

export const seoFormComponentTestData = {
  pluginUID,
  seoId,
  projectCollectionTypes,
  resource: resourceComponentTestData,
};

export const contentTypesListComponentTestData = [
  {
    collectionName,
    fullResults,
    pagination,
    uid,
  },
];

export const translationPickerComponentTestData = {
  translations: trads,
  handleTranslation: () => {},
  value: "",
};

export const contentTypesItemComponentTestData = {
  uid,
  fullResults,
  pagination,
  collectionName,
};
export const previewComponentTestData = {
  selectedSeo,
  pluginUID,
  refreshOptions,
  projectCollectionTypes,
  handleDeleteSeo,
};

export const tableComponentTestData = {
  seos: { collectionName, fullResults, uid, pagination },
  handleDeleteSeo,
  handleEditSeo,
  userEnabledLocales,
  uid,
  defaultLocale,
};
