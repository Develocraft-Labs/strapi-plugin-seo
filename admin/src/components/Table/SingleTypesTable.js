import React, { useCallback, useState, useEffect } from "react";

import { useHistory } from "react-router-dom";
import { GlobalPagination } from "strapi-helper-plugin";
import usePaginationState from "../../hooks/usePaginationState";
import TableItems from "./TableItems";
import { TableComponent, Thead } from "./Table.styles";
import {
  Container,
  TableFooter,
  TableWrap,
  EntryCount,
  TableHeader,
} from "./Table";

/**
 * Component that renders Table.
 * @returns {JSX.Element} Table.
 */
const Table = ({
  items,
  userEnabledLocales,
  uid,
  defaultLocale,
  isSingleType,
  mainField,
}) => {
  const [collectionTypeName] = useState("Single types");
  const [filteredSeos, setFilteredSeos] = useState(items.fullResults);

  const { page, pageSize, count, pageCount, setPage } = usePaginationState({
    currentCount: filteredSeos?.length,
  });

  const history = useHistory();

  const handleParamsChange = useCallback(
    ({ target }) => {
      const { name, value } = target;
      if (name === "params._page") {
        setPage(value);
      }
    },
    [setPage]
  );

  useEffect(() => {
    setFilteredSeos(items.fullResults);
    setPage(1);
  }, [items.fullResults]);

  const getPaginatedData = () => {
    const startIndex = (page - 1) * pageSize;
    const endIndex = startIndex + pageSize;

    return filteredSeos.slice(startIndex, endIndex);
  };

  return (
    <Container>
      <h2>{collectionTypeName}</h2>
      <EntryCount>{`${count} ${
        count > 1 || count === 0 ? "entries" : "entry"
      } found`}</EntryCount>

      <TableWrap>
        <TableComponent>
          <Thead>
            <TableHeader mainField={mainField} />
          </Thead>
          <tbody>
            <TableItems
              paginatedData={getPaginatedData}
              userEnabledLocales={userEnabledLocales}
              uid={uid}
              history={history}
              defaultLocale={defaultLocale}
              isSingleType={isSingleType}
            />
          </tbody>
        </TableComponent>
      </TableWrap>
      {pageCount > 1 && (
        <TableFooter>
          <GlobalPagination
            onChangeParams={handleParamsChange}
            params={{
              _page: page,
              _limit: pageSize,
            }}
            count={count}
          />
        </TableFooter>
      )}
    </Container>
  );
};

export default Table;
