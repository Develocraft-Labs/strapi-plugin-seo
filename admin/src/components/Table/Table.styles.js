import styled, { css } from "styled-components";

export const Thead = styled.thead`
  background: #f3f3f3;
  height: 43px;
  overflow: hidden;
`;

export const TableComponent = styled.table`
  border-radius: 3px;
  border-collapse: initial;
  box-shadow: rgb(227 233 243) 0px 2px 4px;
  color: #212529;
  table-layout: fixed;
  margin-bottom: 0px;
  width: 100%;
`;

export const cellCss = css`
  border: none;
  padding: 0;
  whitespace: nowrap;
`;

export const Cell = styled.tr`
  ${cellCss};
`;

export const ItemRow = styled(Cell)`
  height: 54px;
  background: rgb(255, 255, 255);
  border: none;
  padding: 0rem;
  white-space: nowrap;
`;

export const HeadingCell = styled.th`
  ${cellCss};
  height: 43px;
  font-size: 1.3rem;
  border: none;
  text-transform: capitalize;
  vertical-align: middle;
  padding: 0rem 1.5625rem;
`;
