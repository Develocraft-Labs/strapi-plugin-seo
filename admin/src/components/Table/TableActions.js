import React from "react";
import styled from "styled-components";
import EditTool from "../../ui/Svgs/EditTool";

export const ActionButton = styled.button`
  width: fit-content;
  margin: 0rem 0.8rem;
`;

export const TableActionsWrap = styled.div`
  display: flex;
  justify-content: flex-end;
`;

const TableActions = ({ handleEdit, index }) => {
  return (
    <TableActionsWrap>
      <ActionButton
        onClick={handleEdit}
        data-testid={`edit-seo-button-${index}`}
      >
        <EditTool />
      </ActionButton>
    </TableActionsWrap>
  );
};

export default TableActions;
