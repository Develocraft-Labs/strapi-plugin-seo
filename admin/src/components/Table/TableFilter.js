import React, { useCallback, useState } from "react";
import TextField from "../../ui/Inputs/TextField";
import ArrowUp from "../../ui/Svgs/ArrowUp";
import styled from "styled-components";
import { Row } from "../ui/common";

const HideFilterButton = styled.button`
  width: fit-content;
  padding: 0.3rem 0.7rem;
  align-self: flex-end;
  margin: 1rem 0rem;
  color: rgb(195, 197, 200);
  font-size: 13px;
`;

const FilterButton = styled.button`
  width: fit-content;
  padding: 0.3rem 0.7rem;
  align-self: flex-start;
  margin: 1.8rem 0rem 1rem 0rem;
  background-color: rgb(255, 255, 255);
  border: 1px solid rgb(227, 233, 243);
  border-radius: 2px;
  font-weight: 500;
  font-size: 13px;
`;

const FilterRow = styled(Row)`
  justify-content: space-between;
  padding-bottom: 0rem;
  padding-top: 1.5rem;
`;

const TableFilter = ({ updateTableFilter, filterValue }) => {
  const [showFilter, setShowFilter] = useState(false);

  const handleFilterButton = useCallback(() => {
    setShowFilter((v) => !v);
    updateTableFilter("");
  }, [updateTableFilter]);

  const button = (
    <FilterButton onClick={handleFilterButton}>Filter</FilterButton>
  );

  const handleTableFilter = useCallback(
    (e) => {
      if (e.target.name !== "table filter") {
        return;
      }

      updateTableFilter(e.target.value);
    },
    [updateTableFilter]
  );

  if (!showFilter) {
    return button;
  }

  return (
    <>
      <FilterRow>
        <TextField
          onChange={handleTableFilter}
          name="table filter"
          value={filterValue}
          placeHolder="Title"
          inputWrapperStyles={{
            backgroundColor: "transparent",
            width: "20%",
            alignSelf: "flex-start",
            paddingBottom: "0rem",
            paddingTop: "0rem",
          }}
        />
        <HideFilterButton onClick={handleFilterButton}>
          Hide
          <ArrowUp />
        </HideFilterButton>
      </FilterRow>
      {button}
    </>
  );
};

export default TableFilter;
