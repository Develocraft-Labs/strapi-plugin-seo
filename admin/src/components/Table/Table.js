import React, {
  useCallback,
  useState,
  useEffect,
  useMemo,
  useRef,
} from "react";

import TableFilter from "./TableFilter";
import { useHistory } from "react-router-dom";
import { capitalize } from "lodash";
import { GlobalPagination } from "strapi-helper-plugin";
import usePaginationState from "../../hooks/usePaginationState";
import styled from "styled-components";
import TableItems from "./TableItems";
import { Ellipsis } from "../ui/common";
import { Cell, HeadingCell, TableComponent, Thead } from "./Table.styles";
import { fetchData } from "../../hooks/useContentTypeFilter";

const Container = styled.div`
  position: relative;
  width: 100%;
  max-width: 100%;
  display: flex;
  flex-direction: column;
`;

const TableFooter = styled.div`
  width: fit-content;
  padding: 2rem 0rem 0rem 0rem;
  align-self: flex-end;
`;

const TableWrap = styled.div`
  min-height: 58.5rem;
  padding: 0rem;
`;

const EntryCount = styled.p`
  width: 100%;
  margin: 0px;
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
  font-size: 1.3rem;
  font-weight: 400;
  line-height: normal;
  color: rgb(120, 126, 143);
`;

const getHeaders = ({ mainField }) => [
  "Id",
  mainField || "Title",
  "Created at",
  "Updated at",
  "Content avaliable in",
  "State",
  " ",
];

const TableHeader = ({ mainField }) => {
  const headers = useMemo(() => getHeaders({ mainField }), [mainField]);
  return (
    <Cell>
      {headers.map((heading, index) => {
        return (
          <HeadingCell key={index}>
            <Ellipsis>{heading}</Ellipsis>
          </HeadingCell>
        );
      })}
    </Cell>
  );
};

const handleSingleTypeFilter = async ({
  filterValue,
  mainField,
  apiId,
  collectionName,
  selectedLocale,
  setFilteredItems,
  setIsNotFound,
}) => {
  if (!filterValue) return;

  try {
    const filter = filterValue;
    const data = await fetchData({
      mainField,
      apiId,
      filter,
      collectionName,
      selectedLocale,
    });

    if (data.length == 0) {
      throw new Error("Not Found.");
    }

    setFilteredItems([data]);
    setIsNotFound(false);
  } catch (e) {
    setIsNotFound(true);
    console.error(e);
  }
};

const handleFilter = async ({
  filterValue,
  mainField,
  apiId,
  collectionName,
  selectedLocale,
  setFilteredItems,
  setIsNotFound,
}) => {
  console.log(filterValue, mainField, apiId, collectionName, selectedLocale);
  if (!filterValue) return;

  try {
    const filter = filterValue;
    const data = await fetchData({
      mainField,
      apiId,
      filter,
      collectionName,
      selectedLocale,
    });

    if (data.length == 0) {
      throw new Error("Not Found.");
    }

    setFilteredItems([data]);
    setIsNotFound(false);
  } catch (e) {
    setIsNotFound(true);
    console.error(e);
  }
};

/**
 * Component that renders Table.
 * @returns {JSX.Element} Table.
 */
const Table = ({
  items,
  userEnabledLocales,
  uid,
  defaultLocale,
  isSingleType,
  mainField,
  limit,
  start,
  setStart,
  collectionName,
  pagination,
  apiId,
  selectedLocale,
}) => {
  const [collectionTypeName] = useState(capitalize(collectionName));
  const [filteredItems, setFilteredItems] = useState(items.fullResults);
  const [isNotFound, setIsNotFound] = useState(false);
  const { total, pageSize, pageCount } = pagination;

  const { page, setPage } = usePaginationState({
    currentCount: filteredItems?.length,
  });

  const pageRef = useRef(page);
  const pageRefVal = pageRef.current;
  const [filterValue, setFilterValue] = useState("false");
  const history = useHistory();

  const updateTableFilter = useCallback(
    (value) => {
      setFilterValue(value);
      setPage(1);
    },
    [items.fullResults, setPage]
  );

  const handleParamsChange = useCallback(
    ({ target }) => {
      const { name, value } = target;

      const handleIncrement = () => {
        let increment = 0;

        if (pageRefVal < value) {
          increment = limit + start;
        }
        if (pageRefVal > value) {
          increment = start - limit;
        }

        return increment;
      };

      if (name === "params._page") {
        const increment = handleIncrement();
        const newStart = increment > 0 ? Math.round(increment) : 0;

        pageRef.current = value;
        setStart(newStart);
        setPage(value);
      }
    },
    [setPage, pageRef.current]
  );

  useEffect(() => {
    setFilterValue("");
    setFilteredItems(items.fullResults);
  }, [items.fullResults]);

  const getPaginatedData = useCallback(() => {
    return filteredItems;
  }, [pageRefVal, filteredItems]);

  return (
    <Container>
      <h2>{collectionTypeName}</h2>
      <EntryCount>{`${total} ${
        total > 1 || total === 0 ? "entries" : "entry"
      } found`}</EntryCount>
      <TableFilter
        updateTableFilter={updateTableFilter}
        filterValue={filterValue}
        items={items}
        pageRef={pageRef}
        setStart={setStart}
        isSingleType={isSingleType}
        handleFilter={() =>
          handleFilter({
            filterValue,
            mainField,
            apiId,
            collectionName,
            selectedLocale,
            setFilteredItems,
            setIsNotFound,
          })
        }
        handleSingleTypeFilter={handleSingleTypeFilter}
        setIsNotFound={setIsNotFound}
        setFilteredItems={() => setFilteredItems(items.fullResults)}
      />
      <TableWrap>
        <TableComponent>
          <Thead>
            <TableHeader mainField={mainField} />
          </Thead>
          <tbody>
            <TableItems
              paginatedData={getPaginatedData}
              userEnabledLocales={userEnabledLocales}
              uid={uid}
              history={history}
              defaultLocale={defaultLocale}
              isSingleType={isSingleType}
              isNotFound={isNotFound}
              collectionTypeName={collectionTypeName}
            />
          </tbody>
        </TableComponent>
      </TableWrap>
      {isNotFound
        ? null
        : pageCount > 1 && (
            <TableFooter>
              <GlobalPagination
                onChangeParams={handleParamsChange}
                params={{
                  _page: pageRefVal,
                  _limit: pageSize,
                }}
                count={total}
              />
            </TableFooter>
          )}
    </Container>
  );
};

export default Table;
