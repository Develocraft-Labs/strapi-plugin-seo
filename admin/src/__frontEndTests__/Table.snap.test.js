/**
 * @jest-environment jsdom
 */
import React from "react";
import renderer from "react-test-renderer";

import Table from "../components/Table/Table";
import { tableComponentTestData } from "./testData";

// mock GlobalPagination component from strapi-helper-plugin
jest.mock("strapi-helper-plugin", () => {
  const Pagination = jest.fn(({ children }) => children);
  const GlobalPagination = jest.fn((props) => (
    <Pagination
      onChangeParams={props.handleParamsChange}
      params={props.params}
      count={props.count}
    >
      <div>1, 2</div>
    </Pagination>
  ));
  return { GlobalPagination: GlobalPagination };
});
// mock useParams from react-router
jest.mock("react-router", () => ({
  useHistory: jest.fn().mockReturnValue([]),
  validateInput: () => [],
  InputErrors: () => null,
  LoadingIndicatorPage: () => null,
}));

describe("SNAP TESTING TABlE COMPONENT", () => {
  const {
    handleDeleteSeo,
    handleEditSeo,
    userEnabledLocales,
    uid,
    defaultLocale,
    seos,
  } = tableComponentTestData;

  it("Should Render Successfully", () => {
    const component = renderer.create(
      <Table
        seos={seos}
        handleDeleteSeo={handleDeleteSeo}
        handleEditSeo={handleEditSeo}
        userEnabledLocales={userEnabledLocales}
        uid={uid}
        defaultLocale={defaultLocale}
      />
    );
    const tree = component.toJSON();
    expect(component).toBeDefined();
    expect(tree).toMatchSnapshot();
  });
});
