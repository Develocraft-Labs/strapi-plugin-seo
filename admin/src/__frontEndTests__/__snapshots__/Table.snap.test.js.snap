// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SNAP TESTING TABlE COMPONENT Should Render Successfully 1`] = `
.c8 {
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
  margin: 0;
}

.c2 {
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  padding: 0.3rem 0.7rem;
  -webkit-align-self: flex-start;
  -ms-flex-item-align: start;
  align-self: flex-start;
  margin: 1.8rem 0rem 1rem 0rem;
  background-color: rgb(255,255,255);
  border: 1px solid rgb(227,233,243);
  border-radius: 2px;
  font-weight: 500;
  font-size: 13px;
}

.c11 {
  fill: rgb(109,187,26);
}

.c14 {
  fill: rgba(0,0,0,1);
  height: 1.3rem;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
}

.c13 {
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  margin: 0rem 0.8rem;
}

.c12 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
}

.c5 {
  background: #f3f3f3;
  height: 43px;
  overflow: hidden;
}

.c4 {
  border-radius: 3px;
  border-collapse: initial;
  box-shadow: rgb(227 233 243) 0px 2px 4px;
  color: #212529;
  table-layout: fixed;
  margin-bottom: 0px;
  width: 100%;
}

.c6 {
  border: none;
  padding: 0;
  whitespace: nowrap;
}

.c9 {
  height: 54px;
  background: rgb(255,255,255);
  border: none;
  padding: 0rem;
  white-space: nowrap;
}

.c7 {
  border: none;
  padding: 0;
  whitespace: nowrap;
  height: 43px;
  font-size: 1.3rem;
  border: none;
  vertical-align: middle;
  padding: 0rem 1.5625rem;
}

.c10 {
  height: 53px;
  font-size: 1.3rem;
  line-height: 1.8rem;
  font-weight: 400;
  color: rgb(51,55,64);
  vertical-align: middle;
  border-collapse: collapse;
  border-top: 1px solid rgb(241,241,242);
  padding: 0rem 1.5625rem;
}

.c0 {
  position: relative;
  width: 100%;
  max-width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c15 {
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  padding: 2rem 0rem 0rem 0rem;
  -webkit-align-self: flex-end;
  -ms-flex-item-align: end;
  align-self: flex-end;
}

.c3 {
  min-height: 32rem;
  padding: 0rem;
}

.c1 {
  width: 100%;
  margin: 0px;
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
  font-size: 1.3rem;
  font-weight: 400;
  line-height: normal;
  color: rgb(120,126,143);
}

<div
  className="c0"
>
  <h2>
    Artists
  </h2>
  <p
    className="c1"
  >
    13 entries found
  </p>
  <button
    className="c2"
    onClick={[Function]}
  >
    Filter
  </button>
  <div
    className="c3"
  >
    <table
      className="c4"
    >
      <thead
        className="c5"
      >
        <tr
          className="c6"
        >
          <th
            className="c7"
          >
            <p
              className="c8"
            >
              Id
            </p>
          </th>
          <th
            className="c7"
          >
            <p
              className="c8"
            >
              Title
            </p>
          </th>
          <th
            className="c7"
          >
            <p
              className="c8"
            >
              Created at
            </p>
          </th>
          <th
            className="c7"
          >
            <p
              className="c8"
            >
              Updated at
            </p>
          </th>
          <th
            className="c7"
          >
            <p
              className="c8"
            >
              Content avaliable in
            </p>
          </th>
          <th
            className="c7"
          >
            <p
              className="c8"
            >
              State
            </p>
          </th>
          <th
            className="c7"
          >
            <p
              className="c8"
            >
               
            </p>
          </th>
        </tr>
      </thead>
      <tbody>
        <tr
          className="c6 c9"
          data-testid="collection-item"
        >
          <td
            className="c10"
          >
            1
          </td>
          <td
            className="c10"
          >
            <p
              className="c8"
            >
              title
            </p>
          </td>
          <td
            className="c10"
          >
            <p
              className="c8"
            >
              Tuesday, October 12, 2021 13:49
            </p>
          </td>
          <td
            className="c10"
          >
            <p
              className="c8"
            >
              Friday, October 15, 2021 11:43
            </p>
          </td>
          <td
            className="c10"
          >
            <p
              className="c8"
            >
              English (en)
            </p>
          </td>
          <td
            className="c10"
          >
            <svg
              className="c11"
              height="20"
              viewBox="0 0 24 24"
              width="20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="m10 15.586-3.293-3.293-1.414 1.414L10 18.414l9.707-9.707-1.414-1.414z"
              />
            </svg>
          </td>
          <td
            className="c10"
          >
            <div
              className="c12"
            >
              <button
                className="c13"
                data-testid="edit-seo-button-0"
                onClick={[Function]}
              >
                <svg
                  className="c14"
                  height="36"
                  viewBox="0 0 24 24"
                  width="36"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M19.045 7.401c.378-.378.586-.88.586-1.414s-.208-1.036-.586-1.414l-1.586-1.586c-.378-.378-.88-.586-1.414-.586s-1.036.208-1.413.585L4 13.585V18h4.413L19.045 7.401zm-3-3 1.587 1.585-1.59 1.584-1.586-1.585 1.589-1.584zM6 16v-1.585l7.04-7.018 1.586 1.586L7.587 16H6zm-2 4h16v2H4z"
                  />
                </svg>
              </button>
            </div>
          </td>
        </tr>
        <tr
          className="c6 c9"
          data-testid="collection-item"
        >
          <td
            className="c10"
          >
            2
          </td>
          <td
            className="c10"
          >
            <p
              className="c8"
            >
              title
            </p>
          </td>
          <td
            className="c10"
          >
            <p
              className="c8"
            >
              Tuesday, October 12, 2021 13:49
            </p>
          </td>
          <td
            className="c10"
          >
            <p
              className="c8"
            >
              Friday, October 15, 2021 11:43
            </p>
          </td>
          <td
            className="c10"
          >
            <p
              className="c8"
            >
              English (en)
            </p>
          </td>
          <td
            className="c10"
          >
            <svg
              className="c11"
              height="20"
              viewBox="0 0 24 24"
              width="20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="m10 15.586-3.293-3.293-1.414 1.414L10 18.414l9.707-9.707-1.414-1.414z"
              />
            </svg>
          </td>
          <td
            className="c10"
          >
            <div
              className="c12"
            >
              <button
                className="c13"
                data-testid="edit-seo-button-1"
                onClick={[Function]}
              >
                <svg
                  className="c14"
                  height="36"
                  viewBox="0 0 24 24"
                  width="36"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M19.045 7.401c.378-.378.586-.88.586-1.414s-.208-1.036-.586-1.414l-1.586-1.586c-.378-.378-.88-.586-1.414-.586s-1.036.208-1.413.585L4 13.585V18h4.413L19.045 7.401zm-3-3 1.587 1.585-1.59 1.584-1.586-1.585 1.589-1.584zM6 16v-1.585l7.04-7.018 1.586 1.586L7.587 16H6zm-2 4h16v2H4z"
                  />
                </svg>
              </button>
            </div>
          </td>
        </tr>
        <tr
          className="c6 c9"
          data-testid="collection-item"
        >
          <td
            className="c10"
          >
            3
          </td>
          <td
            className="c10"
          >
            <p
              className="c8"
            >
              title
            </p>
          </td>
          <td
            className="c10"
          >
            <p
              className="c8"
            >
              Tuesday, October 12, 2021 13:49
            </p>
          </td>
          <td
            className="c10"
          >
            <p
              className="c8"
            >
              Friday, October 15, 2021 11:43
            </p>
          </td>
          <td
            className="c10"
          >
            <p
              className="c8"
            >
              English (en)
            </p>
          </td>
          <td
            className="c10"
          >
            <svg
              className="c11"
              height="20"
              viewBox="0 0 24 24"
              width="20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="m10 15.586-3.293-3.293-1.414 1.414L10 18.414l9.707-9.707-1.414-1.414z"
              />
            </svg>
          </td>
          <td
            className="c10"
          >
            <div
              className="c12"
            >
              <button
                className="c13"
                data-testid="edit-seo-button-2"
                onClick={[Function]}
              >
                <svg
                  className="c14"
                  height="36"
                  viewBox="0 0 24 24"
                  width="36"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M19.045 7.401c.378-.378.586-.88.586-1.414s-.208-1.036-.586-1.414l-1.586-1.586c-.378-.378-.88-.586-1.414-.586s-1.036.208-1.413.585L4 13.585V18h4.413L19.045 7.401zm-3-3 1.587 1.585-1.59 1.584-1.586-1.585 1.589-1.584zM6 16v-1.585l7.04-7.018 1.586 1.586L7.587 16H6zm-2 4h16v2H4z"
                  />
                </svg>
              </button>
            </div>
          </td>
        </tr>
        <tr
          className="c6 c9"
          data-testid="collection-item"
        >
          <td
            className="c10"
          >
            4
          </td>
          <td
            className="c10"
          >
            <p
              className="c8"
            >
              title
            </p>
          </td>
          <td
            className="c10"
          >
            <p
              className="c8"
            >
              Tuesday, October 12, 2021 13:49
            </p>
          </td>
          <td
            className="c10"
          >
            <p
              className="c8"
            >
              Friday, October 15, 2021 11:43
            </p>
          </td>
          <td
            className="c10"
          >
            <p
              className="c8"
            >
              English (en)
            </p>
          </td>
          <td
            className="c10"
          >
            <svg
              className="c11"
              height="20"
              viewBox="0 0 24 24"
              width="20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="m10 15.586-3.293-3.293-1.414 1.414L10 18.414l9.707-9.707-1.414-1.414z"
              />
            </svg>
          </td>
          <td
            className="c10"
          >
            <div
              className="c12"
            >
              <button
                className="c13"
                data-testid="edit-seo-button-3"
                onClick={[Function]}
              >
                <svg
                  className="c14"
                  height="36"
                  viewBox="0 0 24 24"
                  width="36"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M19.045 7.401c.378-.378.586-.88.586-1.414s-.208-1.036-.586-1.414l-1.586-1.586c-.378-.378-.88-.586-1.414-.586s-1.036.208-1.413.585L4 13.585V18h4.413L19.045 7.401zm-3-3 1.587 1.585-1.59 1.584-1.586-1.585 1.589-1.584zM6 16v-1.585l7.04-7.018 1.586 1.586L7.587 16H6zm-2 4h16v2H4z"
                  />
                </svg>
              </button>
            </div>
          </td>
        </tr>
        <tr
          className="c6 c9"
          data-testid="collection-item"
        >
          <td
            className="c10"
          >
            5
          </td>
          <td
            className="c10"
          >
            <p
              className="c8"
            >
              title
            </p>
          </td>
          <td
            className="c10"
          >
            <p
              className="c8"
            >
              Tuesday, October 12, 2021 13:49
            </p>
          </td>
          <td
            className="c10"
          >
            <p
              className="c8"
            >
              Friday, October 15, 2021 11:43
            </p>
          </td>
          <td
            className="c10"
          >
            <p
              className="c8"
            >
              English (en)
            </p>
          </td>
          <td
            className="c10"
          >
            <svg
              className="c11"
              height="20"
              viewBox="0 0 24 24"
              width="20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="m10 15.586-3.293-3.293-1.414 1.414L10 18.414l9.707-9.707-1.414-1.414z"
              />
            </svg>
          </td>
          <td
            className="c10"
          >
            <div
              className="c12"
            >
              <button
                className="c13"
                data-testid="edit-seo-button-4"
                onClick={[Function]}
              >
                <svg
                  className="c14"
                  height="36"
                  viewBox="0 0 24 24"
                  width="36"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M19.045 7.401c.378-.378.586-.88.586-1.414s-.208-1.036-.586-1.414l-1.586-1.586c-.378-.378-.88-.586-1.414-.586s-1.036.208-1.413.585L4 13.585V18h4.413L19.045 7.401zm-3-3 1.587 1.585-1.59 1.584-1.586-1.585 1.589-1.584zM6 16v-1.585l7.04-7.018 1.586 1.586L7.587 16H6zm-2 4h16v2H4z"
                  />
                </svg>
              </button>
            </div>
          </td>
        </tr>
      </tbody>
    </table>
  </div>
  <div
    className="c15"
  >
    <div>
      1, 2
    </div>
  </div>
</div>
`;
